### PR TITLE
Fix bugs: (1) infinite loop in e2e tests (2) missing Vote Extensions at `PrepareProposal` time

### PR DESCRIPTION
**Pull request by sergio-mena**
Contributes to #10

While running e2e tests locally, I detected a hard-to-diagnose bug with vote extensions. This PR contains the fix.

Since I did this in isolation (due to remote work), I'll make an effort at explaining the troubleshooting process in this PR. Hence, this PR will be a succession of commits which show the sketch on how I diagnosed and fixed the bug. So, if the reviewer wants to understand what the problem was and how it was solved, the presence of the little ❌ or ✅ next to each commit below is important (you can click and check how the e2e tests failed/passed).

Additionally, I also found that the e2e tests where blocking forever when a node was crashing. This is fixed in the first commit.

* Commit#1: fix the infinite loop in e2e runner that may block the execution indefinitely.
* Commit#2: adapt `ci.toml` to only have one validator, since the bug appears when a recovered node has to propose.
* Commit#3: under the hypothesis that something is wrong with `VoteExtensionsEnableHeight`, we plant a `panic` to have the backtrace if it happens.
  * The enable height in `ci.toml` is set to 1007 from genesis, so we should never return zero in this test case.
* Commit#4: contains the fix. Every time we call `reconstructLastCommit` method of `State`, immediately after we call `updateToState`. Therefore the state object _whose method_ we are calling is incomplete. The fix just queries the _complete_ object, which is passed as a normal parameter (and which will be given to `updateToState` immediately after).
  * Note that, although some checks are failing on that commit (hence the ❌), e2e are now passing
* Commit#5: remove diagnosis-related code, i.e., commit#2 and commit#3

---

#### PR checklist

- [x] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by thanethomson — APPROVED** [review]
Great catch!

File: test/e2e/runner/rpc.go
Comment (cason) [inline test/e2e/runner/rpc.go:1]: No changes in this file, after all? I don't think changing the order of events in a select block has any practical effect.

Comment (sergio-mena) [reply test/e2e/runner/rpc.go:1]: It removes an infinite loop ;-)

Comment (cason) [reply test/e2e/runner/rpc.go:1]: I read the code wrongly, sorry.

Comment (sergio-mena) [reply test/e2e/runner/rpc.go:1]: NP

**Review by cason — APPROVED** [review]

**Pull request merged**